packages/core/src/utils/anthropic-ai/index.ts

```diff
@@ -0,0 +1,242 @@
+      attributes[GEN_AI_REQUEST_MODEL_ATTRIBUTE] = args[0];
+    } else {
+      attributes[GEN_AI_REQUEST_MODEL_ATTRIBUTE] = 'unknown';
+    }
+  }
+
+  return attributes;
+}
+
+/**
+ * Add private request attributes to spans.
+ * This is only recorded if recordInputs is true.
+ */
+function addPrivateRequestAttributes(span: Span, params: Record<string, unknown>): void {
+  if ('messages' in params) {
+    span.setAttributes({ [GEN_AI_REQUEST_MESSAGES_ATTRIBUTE]: JSON.stringify(params.messages) });
+  }
+  if ('input' in params) {
+    span.setAttributes({ [GEN_AI_REQUEST_MESSAGES_ATTRIBUTE]: JSON.stringify(params.input) });
+  }
+  if ('prompt' in params) {
+    span.setAttributes({ [GEN_AI_PROMPT_ATTRIBUTE]: JSON.stringify(params.prompt) });
+  }
+}
+
+/**
+ * Add response attributes to spans
+ */
+function addResponseAttributes(span: Span, response: AnthropicAiResponse, recordOutputs?: boolean): void {
+  if (!response || typeof response !== 'object') return;
+
+  // Private response attributes that are only recorded if recordOutputs is true.
+  if (recordOutputs) {
+    // Messages.create
+    if ('content' in response) {
+      if (Array.isArray(response.content)) {
+        span.setAttributes({
+          [GEN_AI_RESPONSE_TEXT_ATTRIBUTE]: response.content
+            .map((item: { text: string | undefined }) => item.text)
+            .filter((text): text is string => text !== undefined)
+            .join(''),
+        });
+      }
+    }
+    // Completions.create
+    if ('completion' in response) {
+      span.setAttributes({ [GEN_AI_RESPONSE_TEXT_ATTRIBUTE]: response.completion });
+    }
+    // Models.countTokens
+    if ('input_tokens' in response) {
+      span.setAttributes({ [GEN_AI_RESPONSE_TEXT_ATTRIBUTE]: JSON.stringify(response.input_tokens) });
+    }
+  }
+
+  span.setAttributes({
+    [GEN_AI_RESPONSE_ID_ATTRIBUTE]: response.id,
+  });
+  span.setAttributes({
+    [GEN_AI_RESPONSE_MODEL_ATTRIBUTE]: response.model,
+  });
+  if ('created' in response && typeof response.created === 'number') {
+    span.setAttributes({
+      [ANTHROPIC_AI_RESPONSE_TIMESTAMP_ATTRIBUTE]: new Date(response.created * 1000).toISOString(),
+    });
+  }
+  if ('created_at' in response && typeof response.created_at === 'number') {
+    span.setAttributes({
+      [ANTHROPIC_AI_RESPONSE_TIMESTAMP_ATTRIBUTE]: new Date(response.created_at * 1000).toISOString(),
+    });
+  }
+
+  if (response.usage) {
+    setTokenUsageAttributes(
+      span,
+      response.usage.input_tokens,
+      response.usage.output_tokens,
+      response.usage.cache_creation_input_tokens,
+      response.usage.cache_read_input_tokens,
+    );
+  }
+}
+
+/**
+ * Get record options from the integration
+ */
+function getOptionsFromIntegration(): AnthropicAiOptions {
+  const scope = getCurrentScope();
+  const client = scope.getClient();
+  const integration = client?.getIntegrationByName(ANTHROPIC_AI_INTEGRATION_NAME) as AnthropicAiIntegration | undefined;
```

Comment:
You could pass the integration options here as a param as the the place where you use the function has the options available. However, the function is only called when there are no options defined.